### ISSUE TITLE
Inconsistent sync complete toast

### ISSUE THREAD

**Issue by f-odhiambo** [opensrp/fhircore#2942]
**Describe the bug**
After clicking sync(On the registers menu) two toast messages are shown **"Sync complete"** then a second later **"Syncing"**

**To Reproduce**
1. Log in to Bidan, Eir and Cadre
2. Click on the menu on the top left corner
3. Click on sync
4. Two toast messages are shown **Sync complete** the **Syncing** a second later(See attached screen cast)

**Expected behavior**
After clicking on sync the toast message syncing should be shown then later sync complete should show.

https://github.com/ona-health/sid-dto-fhir-resources/assets/73172427/c469525b-257e-4a31-be2d-50abaccff563

**Comment by kelvin-ngure** [comment]
@f-odhiambo
What is happening is intentional due to a missing feature that is coming soon from the SDK. [This SyncJobStatus file will show you the new ENQUEUED state coming soon](https://github.com/google/android-fhir/pull/2142/files#diff-4669dc1484b455bfedfacf6de468f5308717e32364dd9915d6812bbf714311ab)

While there is an ENQUEUED WorkState for scheduled jobs, there is no such sync job status and as a result, when a job is enqueued, the RegisterFragment is instructed to mark it as FINISHED. 

Since every job enters an ENQUEUED phase regardless of whether it is a periodicSync or oneTimeSync, the "Sync Complete" toast shows up immediately the jobs are enqueued. The oneTimeSync jobs quickly jump into STARTED state and so you see the "Syncing" on the UI. Then once the oneTimeSync is complete, the "Sync Complete" toast shows up for the second time

**Comment by kelvin-ngure** [comment]
While looking at this, I spotted what I believe to be a bug or something that can make spotting sync issues harder. When you initially start the app, the sync ends in a FAILED state that is preceded by a GLITCH state. On the UI, the "Sync Completed" toast shows up but then a sync retry begins. The retry is expected since GLITCH should prompt a retry. I will look into the flow and figure out why the "Sync Complete" toast shows up

@f-odhiambo @dubdabasoduba please let me know if we should show a "Sync Failed. Retrying..." toast or something during a Glitch as there is no handler for the GLITCH status

**Comment by ndegwamartin** [comment]
Related ticket here https://github.com/opensrp/fhircore/issues/2947

**Comment by dubdabasoduba** [comment]
@kelvin-ngure  did you stop get the 403 and 502s mentioned on the comment above?

**Comment by kelvin-ngure** [comment]
@ndegwamartin @f-odhiambo from What I see in [the upgrade PR] (https://github.com/opensrp/fhircore/pull/2825). To fix this, we would first need to migrate from using `SyncJobStatus` to using `CurrentSyncJobStatus` which has both work state and SyncJobStatus state representations. Is that migration something we are planning to do?

**Comment by ndegwamartin** [comment]
@kelvin-ngure yeah this needs to be done. Created a ticket here to track this https://github.com/opensrp/fhircore/issues/3059. Feel free to update the ticket with other info needed.

**Comment by ndegwamartin** [comment]
This is still blocked, see https://github.com/google/android-fhir/issues/2472
